lom/src/engine/sequenceHandler.go

```diff
@@ -0,0 +1,832 @@
+    /* Stays in this Q until client reads it */
+    if err := regF.AddServerRequest(action, req); err != nil {
+        return LogError("Internal error: Failed to AddServerRequest (%s)", action)
+    }
+
+    /* Track it in our active requests;  Waits here till response */
+    p.activeRequests[action] = &activeRequest_t {req, uuid, 0}
+
+    /*
+     * NOTE: We only run one timer for seq timeout.
+     *  So a req coming after its timeout, *could* still get consumed.
+     */
+    return nil
+}
+
+
+/* Called upon action de-registration */
+func (p *SeqHandler_t) DropRequest(action string) {
+    if r, ok := p.activeRequests[action]; ok {
+        delete (p.activeRequests, action)
+        if s, ok := p.sequencesByAnomaly[r.anomalyID]; ok {
+            p.completeSequence(s, LoMActionDeregistered, "Action (" + action +") Deregistered")
+            p.resumeNextSequence()
+        }
+    }
+}
+
+
+/* Sort sorted sequences */
+func (p *SeqHandler_t) sortSequences() {
+    sort.Slice(p.sortedSequencesByDue, func(i, j int) bool {
+        return p.sortedSequencesByDue[i].ExpiryEpoch() <
+                        p.sortedSequencesByDue[j].ExpiryEpoch()
+    })
+
+    sort.Slice(p.sortedSequencesByPri, func(i, j int) bool {
+        return (p.sortedSequencesByPri[i].sequence.Priority <
+                        p.sortedSequencesByPri[j].sequence.Priority)
+    })
+}
+
+
+/*
+ * addSequence
+ *
+ * Add this sequence to handler.
+ * Enable timer for seq expiry.
+ * Re-create sorted sequences
+ */
+func (p *SeqHandler_t) addSequence(seq *sequenceState_t, tout int) {
+    if (seq == nil) || (len(seq.context) == 0) {
+        LogError("Expect atleast one context (%v)", seq)
+    } else {
+        p.sequencesByAnomaly[seq.anomalyInstanceId] = seq
+        p.sequencesByFirstAction[seq.context[0].Action] = seq
+        p.sortedSequencesByDue = append(p.sortedSequencesByDue, seq)
+        p.sortedSequencesByPri = append(p.sortedSequencesByPri, seq)
+        m := "Seq: " + seq.sequence.SequenceName + " timeout"
+        seq.expTimer = AddOneShotTimer(int64(tout), m, p.FireTimer)
+        p.sortSequences()
+    }
+}
+
+
+/*
+ * dropSequence
+ *
+ * Remove this sequence from handler.
+ * Disable timer.
+ * Re-create sorted sequences
+ */
+func (p *SeqHandler_t) dropSequence(seq *sequenceState_t) {
+    if (seq == nil) || (len(seq.context) == 0) {
+        LogError("Expect atleast one context (%v)", seq)
+    } else {
+        seq.expTimer.Disable()
+        if p.currentSequence == seq {
+            p.currentSequence = nil
+        }
+        delete (p.sequencesByAnomaly, seq.anomalyInstanceId)
+        delete (p.sequencesByFirstAction, seq.context[0].Action)
+        p.sortedSequencesByDue = make(SortedSequences_t, len(p.sequencesByAnomaly))
+        p.sortedSequencesByPri = make(SortedSequences_t, len(p.sequencesByAnomaly))
+        i := 0
+        for _, v := range(p.sequencesByAnomaly) {
```

Comment:
replace v with sequence ? 
Could you please take care of single letter naming convention in other files as well.

Reply:
Within a small loop, single letter is fine, as the map being walked is explicitly named.
Yet changed.